packages/adapter-cloudflare/index.js

```diff
@@ -301,6 +301,22 @@ function validate_wrangler_config(config_file = undefined) {
 		validate_worker_settings(wrangler_config);
 	}
 
+	const flags = wrangler_config?.compatibility_flags ?? [];
+
+	if (flags.includes('nodejs_compat') && flags.includes('nodejs_als')) {
```

Comment:
We should check that a `main` key is specified in the config too so that it's only needed if a user is deploying a Worker. I've met some folks who are only using the Cloudflare adapter for static sites / single-page apps